src/Sentry.Unity.Android/SentryNativeAndroid.cs

```diff
@@ -132,6 +132,24 @@ public static void Configure(SentryUnityOptions options, ISentryUnityInfo sentry
         options.DiagnosticLogger?.LogInfo("Successfully configured the Android SDK");
     }
 
+    public static void SetTraceId(SentryUnityOptions options)
+    {
+        options.DiagnosticLogger?.LogInfo("Setting Trace ID");
+
+        var traceId = SentrySdk.GetTraceHeader()?.TraceId;
```

Comment:
Ideally we get a hook from the .NET SDK so when a new trace-id is set there, we are able to pass that down to native.

Similar to `ScopeObserver`. Perhaps it should live there actually.

Because we don't want a single trace id for the lifetime of the game ideally. It should recycle "at some point". Perhaps when a scene loads? We could also reset if session ends/restarts.